pkg/virt-api/webhooks/mutating-webhook/mutators/vm-mutator_test.go

```diff
@@ -282,6 +282,34 @@ var _ = Describe("VirtualMachine Mutator", func() {
 		Expect(vmSpec.Template.Spec.Domain.Machine.Type).To(Equal(preference.Spec.Machine.PreferredMachineType))
 	})
 
+	It("should use prefferedStorageClassName", func() {
+		preference := &instancetypev1alpha2.VirtualMachineClusterPreference{
+			ObjectMeta: k8smetav1.ObjectMeta{
+				Name: "machineTypeClusterPreference",
+			},
+			TypeMeta: k8smetav1.TypeMeta{
+				Kind:       apiinstancetype.ClusterSingularPreferenceResourceName,
+				APIVersion: instancetypev1alpha2.SchemeGroupVersion.String(),
+			},
+			Spec: instancetypev1alpha2.VirtualMachinePreferenceSpec{
+				Volumes: &instancetypev1alpha2.VolumePreferences{
+					PreferredStorageClassName: "ceph",
+				},
+			},
+		}
+		_, err := virtClient.VirtualMachineClusterPreference().Create(context.Background(), preference, k8smetav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		vm.Spec.Preference = &v1.PreferenceMatcher{
+			Name: preference.Name,
+		}
+
+		vmSpec, _ := getVMSpecMetaFromResponse()
+		for _, dv := range vmSpec.DataVolumeTemplates {
+			Expect(*dv.Spec.PVC.StorageClassName).To(Equal(preference.Spec.Volumes.PreferredStorageClassName))
+		}
+	})
```

Comment:
As with the functional test I think you're missing the user provided StorageClassName use case here.